### PR TITLE
Add config property for default health group

### LINKED ISSUE #425: Default HealthGroup when not defined
Originally proposed in https://github.com/quarkusio/quarkus/issues/26877.

### PR DESCRIPTION
**Pull request by xstefank**
Fixes #425

### CODE REVIEW

File: implementation/src/main/java/io/smallrye/health/SmallRyeHealthReporter.java [hunk 1]
```diff
@@ -277,8 +283,13 @@ public Uni<SmallRyeHealth> getWellnessAsync() {
     public Uni<SmallRyeHealth> getHealthGroupAsync(String groupName) {
         List<Uni<HealthCheckResponse>> checks = new ArrayList<>();
         if (allHealthChecks != null && allAsyncHealthChecks != null) {
-            initUnis(checks, allHealthChecks.select(HealthGroup.Literal.of(groupName)),
-                    allAsyncHealthChecks.select(HealthGroup.Literal.of(groupName)));
+            if (groupName.equals(defaultHealthGroup)) {
+                initUnis(checks, getHealthChecksWithoutHealthGroup(HealthCheck.class),
```
Comment (jmartisk) [inline implementation/src/main/java/io/smallrye/health/SmallRyeHealthReporter.java:287]: But in this case, if you have health checks that have an explicit `@HealthGroup` with the value equal to the name of the default group, they won't be returned, right? Shouldn't we include them too, along with those that don't have an explicit group defined?

**Comment by jmartisk** [comment]
Updating `health-groups.adoc` a little with this would also be nice :)

**Comment by jmartisk** [comment]
Also sorry for not having this idea right away, but why does a user need a way to define the name of the default group? Can't the name just be hardcoded as `default` (or `_default` or similar)?  It would simply mean "checks that don't have an explicit group". That would be less confusing, I think. 

The way this is designed now, if you want to obtain checks that don't belong to a group, you have to make an additional step as prerequisite - supply a config property whose value is quite irrelevant (and it will probably be something silly like `myAwesomeDefault`), but you still have to supply it.

CC @HerrDerb

**Comment by HerrDerb** [comment]
>  Can't the name just be hardcoded as default?

Yes, it is not absolutly necessary to be able to name the default group, it'd be only some cosmetic beauty.

**Comment by jmartisk** [comment]
Ok then I'd suggest just always considering the name to be `default`. And when somebody requests to view health checks in the `default` group, it would include the ones without an explicit group, as well as potentially those that explicitly declare `@HealthGroup("default")`
But it's up to @xstefank :)

**Comment by xstefank** [comment]
We cannot force all applications to have default health group if no HealthGroup annotation is defined. In that sense, /health/group (returning all HealthGroup checks) would act in the same way as /health (all HealthChecks). So while the name doesn't matter, it provides better flexibility because otherwise we would need a config "enable.default.health.group" which may work for now but if the requirement comes to have a way to change the name then we would need to have two properties.

**Comment by xstefank** [comment]
@jmartisk @HerrDerb so I'm not pushing against your idea but if you think that "default.group.enable" vs "defaul.group.name" property is better I can rework this.

**Comment by HerrDerb** [comment]
@xstefank I'll leave it up to you. My original idea for an explicit "default group name" property, was reasoned as it have more describing value for the individual usecases. In the end I can work with both.

**Comment by xstefank** [comment]
@HerrDerb perfect feedback. Thanks. We agreed with @jmartisk we will go with name property for now.

**Review by HerrDerb — APPROVED** [review]

**Review by jmartisk — APPROVED** [review]

**Pull request merged**